components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue

```diff
@@ -0,0 +1,256 @@
+    <div>
+      <!-- No TTS Voice Warning -->
+      <v-alert
+        v-if="!ttsVoicesExist"
+        class="my-3"
+        type="warning"
+        text
+      >
+        No TTS voices found!
+      </v-alert>
+
+      <!-- TTS Voice Options -->
+      <div
+        v-if="ttsVoicesExist"
+      >
+        <!-- Enable TTS -->
+        <v-switch
+          v-model="useTts"
+          :disabled="!ttsVoicesExist"
+          label="Enable Chat TTS"
+          color="primary"
+          hide-details
+          dense
+          inset
+        />
+
+        <!-- Troll TTS -->
+        <v-switch
+          v-model="trollTts"
+          :disabled="!useTts"
+          label="Allow Troll TTS"
+          color="primary"
+          hide-details
+          dense
+          inset
+        />
+
+        <!-- Read username -->
+        <v-switch
+          v-model="ttsReadUsername"
+          :disabled="!useTts"
+          label="Read Username"
+          color="primary"
+          hide-details
+          dense
+          inset
+          class="mb-3"
+        />
+
+        <v-divider class="mt-3" />
+
+        <!-- TTS Voice Settings Labels -->
+        <v-subheader class="overline mb-0">
+          TTS Voice Settings
+        </v-subheader>
+
+        <!-- TTS Voice -->
+        <v-select
+          v-model="ttsVoice"
+          :items="ttsVoices"
+          :disabled="!useTts"
+          label="TTS Voice"
+          hide-details
+          class="mb-3"
+          outlined
+        />
+
+        <!-- TTS Rate -->
+        <v-slider
+          :disabled="!useTts"
+          label="Speed"
+          v-model="ttsRate"
+          class="align-center"
+          :max="20"
+          :min="1"
+          :step="0.5"
+          hide-details
+        >
+          <template #append>
+            <v-text-field
+              v-model="ttsRate"
+              class="mt-0 pt-0"
+              hide-details
+              single-line
+              type="number"
+              style="width: 50px"
+            />
+          </template>
+        </v-slider>
+
+        <!-- TTS Volume -->
+        <v-slider
+          :disabled="!useTts"
+          label="Vol."
+          v-model="ttsVolume"
+          class="align-center"
+          :max="10"
+          :min="0"
+          :step="0.5"
+          hide-details
+        />
+
+        <!-- TTS Timeout -->
+        <v-slider
+          :disabled="!useTts"
+          label="Timeout"
+          v-model="ttsTimeout"
+          class="align-center"
+          :max="60"
+          :min="0"
+          :step="0.5"
+          hide-details
+        >
+
+          <template #append>
+            <v-text-field
+              v-model="ttsTimeout"
+              class="mt-0 pt-0"
+              hide-details
+              single-line
+              type="number"
+              style="width: 50px"
+            />
+          </template>
+        </v-slider>
+
+        <i class="caption grey--text">note: a value of 0 disables timeout.</i>
```

Comment:
Maybe all settings should have a tooltip explaining them. Not sure what the mobile equivalent would be.

Reply:
Should consider improving documentation all around tbh.

Reply:
My `Chat.vue` diagram is just missing polls :)

Reply:
*dies inside*